cl_dll/vgui_UnicodeTextImage.cpp

```diff
@@ -0,0 +1,558 @@
+
+	virtual void SolveTraverse(VPANEL panel, bool forceApplySchemeSettings = false) = 0;
+	virtual void PaintTraverse(VPANEL panel) = 0;
+
+	virtual void EnableMouseCapture(VPANEL panel, bool state) = 0;
+
+	// returns the size of the workspace
+	virtual void GetWorkspaceBounds(int &x, int &y, int &wide, int &tall) = 0;
+
+	// gets the absolute coordinates of the screen (in windows space)
+	virtual void GetAbsoluteWindowBounds(int &x, int &y, int &wide, int &tall) = 0;
+
+	// gets the base resolution used in proportional mode
+	virtual void GetProportionalBase(int &width, int &height) = 0;
+
+	virtual void CalculateMouseVisible() = 0;
+	virtual bool NeedKBInput() = 0;
+
+	virtual bool HasCursorPosFunctions() = 0;
+	virtual void SurfaceGetCursorPos(int &x, int &y) = 0;
+	virtual void SurfaceSetCursorPos(int x, int y) = 0;
+
+
+	// SRC only functions!!!
+	virtual void DrawTexturedPolygon(VGuiVertex *pVertices, int n) = 0;
+	virtual int GetFontAscent(UnicodeTextImage::HFont font, wchar_t wch) = 0;
+
+	// web browser
+	virtual void SetAllowHTMLJavaScript(bool state) = 0;
+
+	virtual void SetLanguage(const char *pchLang) = 0;
+
+	virtual const char *GetLanguage() = 0;
+
+	virtual bool DeleteTextureByID(int id) = 0;
+
+	virtual void DrawUpdateRegionTextureBGRA(int nTextureID, int x, int y, const unsigned char *pchData, int wide, int tall) = 0;
+
+	virtual void DrawSetTextureBGRA(int id, const unsigned char *pchData, int wide, int tall) = 0;
+
+	virtual void CreateBrowser(VPANEL panel, IHTMLResponses *pBrowser, bool bPopupWindow, const char *pchUserAgentIdentifier) = 0;
+
+	virtual void RemoveBrowser(VPANEL panel, IHTMLResponses *pBrowser) = 0;
+
+	virtual IHTMLChromeController *AccessChromeHTMLController() = 0;
+};
+
+#define VGUI_SURFACE_INTERFACE_VERSION "VGUI_Surface026"
+
+class ILocalize : public IBaseInterface
+{
+public:
+	virtual void Something1() = 0;
+	virtual void Something2() = 0;
+	virtual void Something3() = 0;
+
+	// Converts UTF-8 to WString
+	virtual int ConvertANSIToUnicode(const char *ansi, wchar_t *unicode, int unicodeBufferSizeInBytes) = 0;
+};
+
+#define VGUI_LOCALIZE_INTERFACE_VERSION "VGUI_Localize003"
+
+bool g_bSurfaceLoaded = false;
+
+// hw.dll or sw.dll on Windows, hw.so on Linux, hw.dylib on macOS.
+CSysModule *g_hEngineModule = nullptr;
+CSysModule *g_hVGuiModule = nullptr;
+
+ISurface *g_pVGuiSurface = nullptr;
+ILocalize *g_pVGuiLocalize = nullptr;
+
+}
+
+void UnicodeTextImage::initInterfaces()
+{
+	static bool bTriedToLoad = false;
+	if (bTriedToLoad)
+		return;
+
+	// Surface needs to be loaded only once, even if it fails.
+	bTriedToLoad = true;
+
+#ifdef _WIN32
+
+	HMODULE hw, sw, vgui2;
+
+	// Try hardware engine (hw.dll)
+	GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "hw.dll", &hw);
+	if (hw)
+	{
+		g_hEngineModule = reinterpret_cast<CSysModule *>(hw);
+	}
+	else
+	{
+		// Try software engine (sw.dll)
+		GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "hw.dll", &sw);
```

Comment:
the classic copy-paste
```suggestion
		GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "sw.dll", &sw);
```